pageserver/src/http/routes.rs

```diff
@@ -114,7 +114,10 @@ fn local_timeline_info_from_loaded_timeline(
         prev_record_lsn: Some(timeline.get_prev_record_lsn()),
         latest_gc_cutoff_lsn: *timeline.get_latest_gc_cutoff_lsn(),
         timeline_state: LocalTimelineState::Loaded,
-        current_logical_size: Some(timeline.get_current_logical_size()),
+        current_logical_size: timeline
+            .get_logical_size_if_initialized()
```

Comment:
Before, we used to return some odd logical size there: if no `init_logical_size` was called, we would return the intermediate increments added after walreceivers' flush/commit calls.
After it was called, we've started sending something close to reality.


Now, we send `None` if no init was done before: we do send back this `TimelineInfo` on every timeline creation/branching, so always calling init here slows things drastically.
After the init is called, we start to send the "proper" size, meanwhile `current_logical_size_non_incremental` is available for the whole size if needed.